Userland/Libraries/LibWeb/CSS/FontFaceSet.cpp

```diff
@@ -97,8 +97,33 @@ FontFaceSet::add(JS::Handle<FontFace> face)
 // https://drafts.csswg.org/css-font-loading/#dom-fontfaceset-delete
 bool FontFaceSet::delete_(JS::Handle<FontFace> face)
 {
-    // FIXME: Do the actual spec steps
-    return m_set_entries->set_remove(face);
+    // 1. If font is CSS-connected, return false and exit this algorithm immediately.
+    if (face->is_css_connected()) {
+        return false;
+    }
+
+    // 2. Let deleted be the result of removing font from the FontFaceSet’s set entries.
+    bool deleted = m_set_entries->set_remove(face);
+
+    // 3. If font is present in the FontFaceSet’s [[LoadedFonts]], or [[FailedFonts]] lists, remove it.
+    if (auto it = m_loaded_fonts.find_if([face](auto const& entry) { return entry == face; }); it != m_loaded_fonts.end()) {
+        m_loaded_fonts.remove(it.index());
+    }
```

Comment:
using remove_first_matching() instead would make it look simpler

Reply:
ah cool, I'll change it to that method :)

Reply:
looks much cleaner now.